src/objective.jl

```diff
@@ -285,9 +285,7 @@ function add_objective!(connection, model, variables, expressions, model_paramet
 
     units_on_cost = @expression(
         model,
-        sum(
-            row.cost * units_on for (row, units_on) in zip(indices, variables[:units_on].container)
-        )
+        sum(row.cost * variables[:units_on].container[row.id] for row in indices)
     )
```

Comment:
Indeed, the `indices` table has a `WHERE` condition, reducing the number of rows and creating a mismatch with `variables[:units_on]`. Thanks for catching this @urosgluscevic.

The solution is correct, but I will add `var_units_on = variables[:units_on].container` to avoid doing the lookup in the loop.